lib/Passes/CodeQualityStatCollector.cpp

```diff
@@ -57,32 +59,61 @@ class ConditionalComplexityVisitor
 
 
 llvm::PreservedAnalyses
-CodeQualityStatCollector::run(llvm::Function &function,
-                              llvm::FunctionAnalysisManager &analysisManager) {
+CodeQualityStatCollector::run(llvm::Module &module,
+                              llvm::ModuleAnalysisManager &analysisManager) {
   ConditionalComplexityVisitor complexity_visitor;
-  for (auto &i : llvm::instructions(function)) {
-    if (auto *int_to_ptr = llvm::dyn_cast<llvm::IntToPtrInst>(&i)) {
-      IntToPointerCasts++;
-    }
+  llvm::GlobalVariable* anvill_sp = module.getGlobalVariable(kSymbolicSPName);
+  llvm::GlobalVariable* anvill_pc = module.getGlobalVariable(kSymbolicPCName);
+
+  llvm::DenseSet<const llvm::Function*> sp_funcs;
+  llvm::DenseSet<const llvm::Function*> pc_funcs;
 
-    if (auto *int_to_ptr = llvm::dyn_cast<llvm::PtrToIntInst>(&i)) {
-      PointerToIntCasts++;
+  if (anvill_sp != nullptr) {
+    for (const auto &U: anvill_sp->uses()) {
+      const auto &user = U.getUser();
+      if (const llvm::Instruction *I = llvm::dyn_cast<llvm::Instruction>(user)) {
+        sp_funcs.insert(I->getFunction());
+      }
     }
+  }
 
+  AnvillStackPointers = sp_funcs.size();
```

Comment:
As a pass, this probably isn't safe since this can be run on multiple modules, should probably be an add. Not the end of the world though